### PR TITLE
chore: use default node cache

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
chore

**Did you add tests for your changes?**
NA

**If relevant, did you update the documentation?**
NA

**Summary**
- Use inbuilt node cache
- 
**Does this PR introduce a breaking change?**
no

**Other information**

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -70,17 +70,7 @@ jobs:
...
-        id: cache
-        uses: actions/cache@v2
-        with:
-          path: |
-            node_modules
-            */*/node_modules
-          key: c-${{ runner.os }}-${{ matrix.webpack-version }}-yarn-${{ hashFiles('**/yarn.lock', './yarn.lock') }}
-          restore-keys: |
-            c-${{ runner.os }}-${{ matrix.webpack-version }}-yarn-
+          cache: 'yarn'
```
Comment (alexander-akait) [inline .github/workflows/nodejs.yml:73]: We need better key here, otherwise cache was broken

Comment (alexander-akait) [reply .github/workflows/nodejs.yml:73]: we need here `webpack-version` and `hashFiles`

Comment (anshumanv) [reply .github/workflows/nodejs.yml:73]: yep, I don't see it on the official docs, lemme see if something is possible

Comment (anshumanv) [reply .github/workflows/nodejs.yml:73]: The action computes a hash internally so we won't need hashFiles

**Review by snitin315 — APPROVED** [review]

**Review by snitin315 — COMMENTED** [review]
One more instance -

https://github.com/webpack/webpack-cli/blob/53622dc74b8a8c4289b4f4e273414a401261ae67/.github/workflows/nodejs.yml#L33

**Comment by anshumanv** [comment]
no need cache for lint job I think

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (alexander-akait) [reply .github/workflows/nodejs.yml]: Can you provide line where it calculated?

Comment (anshumanv) [reply .github/workflows/nodejs.yml]: yep sorry should've done earlier - https://github.com/actions/setup-node/blob/e215578cb0305f14eb5d11939e0f4610ae5bf7e6/src/cache-restore.ts#L26

**Review by alexander-akait — APPROVED** [review]
/cc @webpack/cli-team

**Review by snitin315 — APPROVED** [review]

**Pull request merged**